src/general/NewGameSettings.cs

```diff
@@ -0,0 +1,676 @@
+        playerDeathPopulationPenalty = GetNode<HSlider>(PlayerDeathPopulationPenaltyPath);
+        playerDeathPopulationPenaltyReadout = GetNode<LineEdit>(PlayerDeathPopulationPenaltyReadoutPath);
+        glucoseDecayRate = GetNode<HSlider>(GlucoseDecayRatePath);
+        glucoseDecayRateReadout = GetNode<LineEdit>(GlucoseDecayRateReadoutPath);
+        osmoregulationMultiplier = GetNode<HSlider>(OsmoregulationMultiplierPath);
+        osmoregulationMultiplierReadout = GetNode<LineEdit>(OsmoregulationMultiplierReadoutPath);
+        freeGlucoseCloudButton = GetNode<Button>(FreeGlucoseCloudButtonPath);
+        mapTypeButton = GetNode<OptionButton>(MapTypeButtonPath);
+        lifeOriginButton = GetNode<OptionButton>(LifeOriginButtonPath);
+        lifeOriginButtonAdvanced = GetNode<OptionButton>(LifeOriginButtonAdvancedPath);
+        lawkButton = GetNode<Button>(LAWKButtonPath);
+        lawkAdvancedButton = GetNode<Button>(LAWKAdvancedButtonPath);
+        gameSeed = GetNode<LineEdit>(GameSeedPath);
+        gameSeedAdvanced = GetNode<LineEdit>(GameSeedAdvancedPath);
+        includeMulticellularButton = GetNode<Button>(IncludeMulticellularButtonPath);
+        easterEggsButton = GetNode<Button>(EasterEggsButtonPath);
+        confirmButton = GetNode<Button>(ConfirmButtonPath);
+
+        mpMultiplier.MinValue = Constants.MIN_MP_MULTIPLIER;
+        mpMultiplier.MaxValue = Constants.MAX_MP_MULTIPLIER;
+        aiMutationRate.MinValue = Constants.MIN_AI_MUTATION_RATE;
+        aiMutationRate.MaxValue = Constants.MAX_AI_MUTATION_RATE;
+        compoundDensity.MinValue = Constants.MIN_COMPOUND_DENSITY;
+        compoundDensity.MaxValue = Constants.MAX_COMPOUND_DENSITY;
+        playerDeathPopulationPenalty.MinValue = Constants.MIN_PLAYER_DEATH_POPULATION_PENALTY;
+        playerDeathPopulationPenalty.MaxValue = Constants.MAX_PLAYER_DEATH_POPULATION_PENALTY;
+        glucoseDecayRate.MinValue = Constants.MIN_GLUCOSE_DECAY * 100;
+        glucoseDecayRate.MaxValue = Constants.MAX_GLUCOSE_DECAY * 100;
+        osmoregulationMultiplier.MinValue = Constants.MIN_OSMOREGULATION_MULTIPLIER;
+        osmoregulationMultiplier.MaxValue = Constants.MAX_OSMOREGULATION_MULTIPLIER;
+
+        settings = new WorldGenerationSettings();
+        difficultyPresets = SimulationParameters.Instance.GetAllDifficultyPresets();
+        normal = SimulationParameters.Instance.GetDifficultyPreset("normal");
+        custom = SimulationParameters.Instance.GetDifficultyPreset("custom");
+
+        foreach (DifficultyPreset preset in difficultyPresets)
+        {
+            difficultyPresetButton.AddItem(preset.Name);
+            difficultyPresetAdvancedButton.AddItem(preset.Name);
+        }
+
+        // Do this in case default values in NewGameSettings.tscn don't match the normal preset
+        InitialiseToPreset(normal);
+
+        var seed = GenerateNewRandomSeed();
+        gameSeed.Text = seed;
+        gameSeedAdvanced.Text = seed;
+        SetSeed(seed);
```

Comment:
Why doesn't SetSeed update gameSeed and gameSeedAdvanced automatically (assuming the given new seed is valid)?